File: YuGiOh2/Cards/C53129443.cs
﻿using System.Linq;
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 黑洞
    /// </summary>
    public class C53129443
    {
        public static int Type { get; } = (int)ChooseTargetType.None;

        public static bool CheckIfAvailable(Player player)
        {
            return player.Enemy.Field.MonsterFields.Any(c => c != null) ||
                player.Field.MonsterFields.Any(c => c != null);
        }

        public static void ProcessEffect(Player player)
        {
            Destroy(player);
            Destroy(player.Enemy);
        }

        private static void Destroy(Player player)
        {
            for (int i = 0; i < 5; i++)
            {
                if (player.Field.MonsterFields[i] == null)
                    continue;

                player.AddCardToGrave(ref player.Field.MonsterFields[i]);
                player.Field.MonsterFields[i] = null;
            }
        }
    }
}

File: YuGiOh2/Scripts/C53129443.lua
Type = 0;
--黑洞
function checkIfAvailable(player)
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil then
            return true;
        end
    end
    for i, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil then
            return true;
        end
    end
    return false;
end

function processEffect(player)
    destroy(player.Enemy);
    destroy(player);
end

function destroy(player)
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil then
            player:AddCardToGrave(monster);
            player:ClearMonsterField(i - 1);
        end
    end
end